webapp/src/components/github_milestone_selector/github_milestone_selector.jsx

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import IssueAttributeSelector from 'components/issue_attribute_selector';
+
+export default class GithubMilestoneSelector extends PureComponent {
+    static propTypes = {
+        repo: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        selectedMilestone: PropTypes.number.isRequired,
+        onChange: PropTypes.func.isRequired,
+        milestones: PropTypes.array,
+        actions: PropTypes.shape({
+            getMilestones: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    loadMilestones = async () => {
+        if (this.props.repo === '') {
+            return [];
+        }
+
+        const options = await this.props.actions.getMilestones(this.props.repo);
+
+        if (options.error) {
+            throw new Error('Failed to load milestones');
+        }
+
+        if (!options || !options.data) {
+            return [];
+        }
+
+        return options.data.map((option) => ({
+            value: option.title,
+            label: option.title,
+        }));
+    };
+
+    onChange = (selection) => {
+        if (!this.props.milestones) {
+            this.props.onChange(0);
+            return;
+        }
+
+        // we have to find the selected milestone from the options in order to insert its number
+        const milestone = this.props.milestones.find((m) => m.title === selection);
+        this.props.onChange(milestone ? milestone.number : 0);
+    }
+
+    render() {
+        let milestone = '';
+        if (this.props.selectedMilestone > 0 && this.props.milestones) {
+            milestone = this.props.milestones.find((m) => m.number === this.props.selectedMilestone).title;
+        }
+
+        return (
+            <div className='form-group margin-bottom x3'>
+                <label className='control-label margin-bottom x2'>
+                    {'Milestone'}
+                </label>
+                <IssueAttributeSelector
+                    {...this.props}
+                    isMulti={false}
+                    selection={milestone}
+                    loadOptions={this.loadMilestones}
+                    onChange={this.onChange}
+                />
```

Comment:
Non blocking idea: 

If no repository is selected, `No Options` will be shown. Is it possible to change to text to `Please select a repository first`
![Screenshot from 2020-08-07 09-54-07](https://user-images.githubusercontent.com/16541325/89622991-2f630780-d894-11ea-9d29-86b293e8a241.png)